### ISSUE TITLE
Is it possible to make an entire form read-only?   

### ISSUE THREAD

**Issue by Thorski** [dschnelldavis/angular2-json-schema-form#97]
Use case - using schema form for editing objects.   Some users cannot
edit these objects but should be able to view them.

Another use case - some objects are predefined in the system and cannot be edited,
would like to use the same schema and layout for these objects.

Would like to set something on the form via script or binding where I can make the whole
form read-only including disabling the submit button.

Is this possible?

Thanks.

**Comment by Thorski** [comment]
I made this happen by putting
`  <fieldset [disabled]="item.readonly">
     <json-schema-form [form]="itemForm".... >
     </json-schema-form>
   </fieldset>`

**Comment by Thorski** [comment]
I made this happen by putting a fieldset wrapper around the json schema form:
`          <fieldset [disabled]="item.readonly">
              <json-schema-form [form]="itemForm" ... >
          </fieldset>`

**Comment by dschnelldavis** [comment]
Yes, there's a fairly straightforward way to do this. Just add `"readonly": true` to your defaultOptions object, which will make it the default for every widget.

In your site, set angular2-json-schema-form's `options` input to the following:
```javascript
{
  "defaultOptions": { "readonly": true }
}
```

Or, in the [demonstration playground](https://angular2-json-schema-form.firebaseapp.com/) (which uses the combined `form` input), to make any schema read-only, add the following, starting on the _second_ line:
```javascript
  "options": {
    "defaultOptions": { "readonly": true }
  },
```

**Comment by catull** [comment]
@dschnelldavis Did you mean like this?

It does not work, though, I was able to modify attributes, look for "(changed)" in them.

After I pasted in the "options" snippet, I selected the whole field, removed the content and pasted it back.

Still no change, was able to modify whichever field I stepped on.

![screen shot 2018-03-14 at 11 59 28](https://user-images.githubusercontent.com/150055/37398571-3f95301a-277f-11e8-8006-460a6e0aa4e6.png)
